include/drivers/kscan.h

```diff
@@ -128,6 +167,95 @@ static inline int z_impl_kscan_disable_callback(const struct device *dev)
 	return api->disable_callback(dev);
 }
 
+/**
+ * @brief Configure a Keyboard matrix scan instance.
+ *
+ * @param dev Pointer to the device structure for the driver instance.
+ * @param config Pointer to a kscan_matrix_config structure instance
+ * event such as key pressed/released.
+ *
+ * @retval 0 If successful.
+ * @retval Negative errno code if failure.
+ */
+__syscall int kscan_config_matrix(const struct device *dev, struct kscan_matrix_config *config);
+
+static inline int z_impl_kscan_config_matrix(const struct device *dev,
+					     struct kscan_matrix_config *config)
+{
+	const struct kscan_driver_api *api = (struct kscan_driver_api *)dev->api;
+
+	if (api->config_matrix == NULL) {
+		return -ENOTSUP;
```

Comment:
`-ENOSYS`

Reply:
same for others ^

Reply:
Done by replacing all ENPTSUP with -ENOSYS.